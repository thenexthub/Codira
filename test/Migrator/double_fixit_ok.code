// RUN: %empty-directory(%t)
// RUN: not %target-language-frontend -typecheck -update-code -primary-file %s -emit-migrated-file-path %t/double_fixit_ok.result -language-version 4 -parse-as-library
// RUN: %diff -u %s.expected %t/double_fixit_ok.result
// RUN: %target-language-frontend -typecheck %s.expected -language-version 5 -parse-as-library

@available(language, obsoleted: 4, renamed: "Thing.constant___renamed")
immutable ThingConstantGotRenamed = 1

@available(language, obsoleted: 4, renamed: "Thing.constant_renamed")
immutable ThingConstantWasRenamed = 1

struct Thing {
  static immutable constant___renamed = 1
  static immutable constant_renamed = 1
  fn foo(_ c: Int) {}
}

class MyClass {
  fn foo() {
    immutable _: Thing = {
      immutable t = Thing()
      t.foo(ThingConstantGotRenamed)
      t.foo(ThingConstantWasRenamed)
      return t
    }()
  }
}


