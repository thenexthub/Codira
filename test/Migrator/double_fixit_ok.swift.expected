// RUN: %empty-directory(%t)
// RUN: not %target-language-frontend -typecheck -update-code -primary-file %s -emit-migrated-file-path %t/double_fixit_ok.result -language-version 4 -parse-as-library
// RUN: %diff -u %s.expected %t/double_fixit_ok.result
// RUN: %target-language-frontend -typecheck %s.expected -language-version 5 -parse-as-library

@available(language, obsoleted: 4, renamed: "Thing.constant___renamed")
let ThingConstantGotRenamed = 1

@available(language, obsoleted: 4, renamed: "Thing.constant_renamed")
let ThingConstantWasRenamed = 1

struct Thing {
  static let constant___renamed = 1
  static let constant_renamed = 1
  fn foo(_ c: Int) {}
}

class MyClass {
  fn foo() {
    let _: Thing = {
      let t = Thing()
      t.foo(Thing.constant___renamed)
      t.foo(Thing.constant_renamed)
      return t
    }()
  }
}


